paper-api/src/main/java/org/bukkit/structure/Structure.java

```diff
@@ -93,6 +93,26 @@ public interface Structure extends PersistentDataHolder {
      */
     void place(@NotNull Location location, boolean includeEntities, @NotNull StructureRotation structureRotation, @NotNull Mirror mirror, int palette, float integrity, @NotNull Random random);
 
+    /**
+     * Place a structure in the world.
+     *
+     * @param location The location to place the structure at.
+     * @param includeEntities If the entities present in the structure should be
+     * spawned.
+     * @param structureRotation The rotation of the structure.
+     * @param mirror The mirror settings of the structure.
+     * @param palette The palette index of the structure to use, starting at
+     * {@code 0}, or {@code -1} to pick a random palette.
+     * @param integrity Determines how damaged the building should look by
+     * randomly skipping blocks to place. This value can range from 0 to 1. With
+     * 0 removing all blocks and 1 spawning the structure in pristine condition.
+     * @param random The randomizer used for setting the structure's
+     * @param strict Whether to prevent block updates (ie: prevent flowing water or fence connections)
+     * {@link org.bukkit.loot.LootTable}s and integrity.
+     */
+    @ApiStatus.Experimental
+    void place(@NotNull Location location, boolean includeEntities, @NotNull StructureRotation structureRotation, @NotNull Mirror mirror, int palette, float integrity, @NotNull Random random, boolean strict);
```

Comment:
Is it worth having some sort of `PlacementOptions` object instead of having multiple different methods with ~10 arguments each?

Reply:
Who doesn't love 10 arguments 😬 ... jk, but yeah that might be a better idea.

Reply:
Update: 
Ok, I made a placement class. 
I haven't filled in the Javadocs yet. I wanted to get opinions on how I made everything before I filled in javadocs.
(ie: just want to make sure the structure is what Paper would expect)

Reply:
At first I didn't think that would make sense, but yeah... it does make sense.
I have moved them 👍🏻